core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -83,7 +83,12 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
    *
    * It is also only available after [[init]] is invoked.
    */
-  def fetchBlockSync(host: String, port: Int, execId: String, blockId: String): ManagedBuffer = {
+  def fetchBlockSync(
+      host: String,
+      port: Int,
+      execId: String,
+      blockId: String,
+      timeout: Int): ManagedBuffer = {
```

Comment:
Maybe we should put all these configurations into a single file, which is
more clear.

2015-03-26 15:59 GMT+08:00 Shixiong Zhu <notifications@github.com>:

> In core/src/main/scala/org/apache/spark/network/BlockTransferService.scala
> <https://github.com/apache/spark/pull/5197#discussion_r27193772>:
>
> > @@ -83,7 +83,12 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
> >     *
> >     * It is also only available after [[init]] is invoked.
> >     */
> > -  def fetchBlockSync(host: String, port: Int, execId: String, blockId: String): ManagedBuffer = {
> > +  def fetchBlockSync(
> > +      host: String,
> > +      port: Int,
> > +      execId: String,
> > +      blockId: String,
> > +      timeout: Int): ManagedBuffer = {
>
> I have a question about the time unit of such configurations. Do you think
> it's better to use same time unit for all configurations? Currently, it's
> hard to know the time unit when reading a configuration file. I usually
> need to search them in the codes or docs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/5197/files#r27193772>.
>